src/hotspot/share/opto/compile.cpp

```diff
@@ -698,6 +698,7 @@ Compile::Compile( ciEnv* ci_env, ciMethod* target, int osr_bci,
   }
   set_print_inlining(directive->PrintInliningOption || PrintOptoInlining);
   set_print_intrinsics(directive->PrintIntrinsicsOption);
+  set_pea_method_only(directive->PEAMethodOnlyOption);
```

Comment:
This change has subtle difference from PEAMethodOnly. 

Because we do PEA in C2 parse-time, it's inter-procedural regarding of inlined methods. if you think of an inline-tree, there are multiple methods and top-level method is the root node of the tree, isn't it? 

If you set PEAMethodOnlyOption like this, we only match the top-level method. We can't match an inlined method. It was pretty useful to match a specific method in the inline tree. 

I don't see an easy way to support multiple entries of PEAMethodOnly. if you really need to match multiple methods, how about we have both? you can rename the option and keep it.

Reply:
I brought back -XX:+PEAMethodOnly in the newest commit

I'm not exactly sure how the compile commands work. I would think that inlined methods are part of the same compilation unit, and compile commands are set per compilation unit. So my guess would be that the compile command would work for inlined methods as well.

Reply:
I think my changes are wrong actually. The `||` condition makes us compile all the time, since `-XX:+PEAMethodOnly` is on by default. I'll look into it.

Reply:
'PEAMethodOnly' was an ccstr. 

You added  'PEAMethodOnly' to CompileDirective. It actually is PEAMethodOnlyOption in global. the type of PEAMethodOnlyOption is boolean. 


for every compilation unit, PEAMethodOnlyOption is evaluated. if the top-level method does match one criterion, that option is set. that's why I said your approach can only match top-level method. 

The reason Compiler directive can support multiple entries is that it maintains a stack. those method matchers form a daisy chain. 

I think your code works. the only confusion is that names. how about you rename it 'PEAMethodOnlyTop'?